core/src/main/java/com/google/googlejavaformat/java/Main.java

```diff
@@ -142,28 +143,45 @@ private int formatFiles(CommandLineOptions parameters, JavaFormatterOptions opti
       } catch (ExecutionException e) {
         if (e.getCause() instanceof FormatterException) {
           for (FormatterDiagnostic diagnostic : ((FormatterException) e.getCause()).diagnostics()) {
-            errWriter.println(result.getKey() + ":" + diagnostic.toString());
+            errWriter.println(path + ":" + diagnostic.toString());
           }
         } else {
-          errWriter.println(result.getKey() + ": error: " + e.getCause().getMessage());
+          errWriter.println(path + ": error: " + e.getCause().getMessage());
           e.getCause().printStackTrace(errWriter);
         }
         allOk = false;
         continue;
       }
+      boolean idempotent = formatted.equals(inputs.get(path)); // not changed
+      if (parameters.isValidationMode()) {
+        if (idempotent) {
+          continue; // no formatting needed, validate next
+        }
+        errWriter.println(path + ": needs formatting");
```

Comment:
I'd expect stdout for this, and dropping `: needs formatting` would make this easier to use in scripts.

Reply:
Ok.